### ISSUE TITLE
[Rust] Implement new Concept Exercise: structs

### ISSUE THREAD

**Issue by efx** [exercism/v3#1130]
This issue describes how to implement the `structs` concept exercise for the rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of working with named-field `structs` in Rust. `structs` are an essential user defined type.

## Learning objectives

- Know what the `struct` keyword means 
- Know how to create the `struct` keyword means 
- Know that you can define methods on structs
- Know of the basics of different kinds of structs

## Out of scope

- representation in memory
- advanced treatment of
  - generic structs
  - unit structs (Zero-Size structs)
  - tuple structs

## Concepts

- structs

## Prerequisites

- basics
- string-use
- unsigned-ints
- floats

### Hints

- https://doc.rust-lang.org/book/ch05-00-structs.html
- https://doc.rust-lang.org/stable/rust-by-example/custom_types/structs.html

### After

- https://doc.rust-lang.org/book/ch05-00-structs.html
- https://doc.rust-lang.org/stable/rust-by-example/custom_types/structs.html

## Representer

No changes required

## Analyzer

No changes required

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/reference/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.  In addition you may want to make use of the shell script while in the `concept` directory, with the arg loops, e.g. `sh boil_exercise.sh vec`

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
